hibernate-core/src/main/java/org/hibernate/internal/CacheImpl.java

```diff
@@ -39,7 +42,10 @@
 	private final transient RegionFactory regionFactory;
 	private final transient UpdateTimestampsCache updateTimestampsCache;
 	private final transient ConcurrentMap<String, QueryCache> queryCaches;
-	private final transient ConcurrentMap<String, Region> allCacheRegions = new ConcurrentHashMap<String, Region>();
+	private final transient ConcurrentMap<String, EntityRegion> entityRegionMap = new ConcurrentHashMap<String, EntityRegion>();
+	private final transient ConcurrentMap<String, CollectionRegion> collectionRegionMap = new ConcurrentHashMap<String, CollectionRegion>();
```

Comment:
@Sanne @rvansa any memory consumption concerns about this PR adding three ConcurrentHashMaps + three HashMaps per SessionFactory?

Reply:
Nope, this does not cause any allocations during runtime. And users don't have thousands of regions.